hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java

```diff
@@ -1337,7 +1352,11 @@ private void hedgedFetchBlockByteRange(LocatedBlock block, long start,
         } catch (InterruptedException ie) {
           // Ignore and retry
         }
-        if (refetch) {
+        // if refetch is true then all nodes are in deadlist or ignorelist
+        // we should loop through all futures and remove them so we do not
```

Comment:
Could you add punctuation here and start new sentences with caps. It was hard to read the flow of the comment.

What is the deadlist?

Reply:
fixed comments. deadlist is actually deadNodes (I fixed that comment as well.)
When connections fail (in both hedged and non hedged code path) nodes are added to the deadNodes collection to try other nodes. Once `chooseDataNode` returns `null` (or more accurately `getBestNodeDNAddrPair`) it calls `refetchLocations` which clears the deadNodes `clearLocalDeadNodes()` and now with my change, also clears the ignore list. 

Note we have added an assumption to this method `refetchLocations`. The comment I added to `refetchLocations`
``` 
 /**
   * RefetchLocations should only be called when there are no active requests
   * to datanodes. In the hedged read case this means futures should be empty
   */
   ```